mpas_analysis/ocean/conservation.py

```diff
@@ -0,0 +1,633 @@
+                legendText.append(legend_text)
+                lineColors.append(config.get('conservation', 'controlColor'))
+                lineStyles.append(lineStylesBase[index])
+
+        lineWidths = [3 for i in fields]
+        if config.has_option('conservation', 'movingAveragePoints'):
+            movingAveragePoints = config.getint('conservation',
+                                                'movingAveragePoints')
+        else:
+            movingAveragePoints = None
+
+        if config.has_option('conservation', 'firstYearXTicks'):
+            firstYearXTicks = config.getint('conservation',
+                                            'firstYearXTicks')
+        else:
+            firstYearXTicks = None
+
+        if config.has_option('conservation', 'yearStrideXTicks'):
+            EearStrideXTicks = config.getint('conservation',
+                                             'yearStrideXTicks')
+        else:
+            yearStrideXTicks = None
+
+        timeseries_analysis_plot(config, fields, calendar=self.calendar,
+                                 title=title, xlabel=xLabel, ylabel=yLabel,
+                                 movingAveragePoints=movingAveragePoints,
+                                 lineColors=lineColors,
+                                 lineStyles=lineStyles[:len(fields)],
+                                 lineWidths=lineWidths,
+                                 legendText=legendText,
+                                 firstYearXTicks=firstYearXTicks,
+                                 yearStrideXTicks=yearStrideXTicks)
+
+        # save the plot to the output file
+        plt.savefig(outFileName)
+
+        # here's an example of how you would create an XML file for this plot
+        # with the appropriate entries.  Some notes:
+        # * Gallery groups typically represent all the analysis from a task,
+        #   or sometimes from multiple tasks
+        # * A gallery might be for just for one set of observations, one
+        #   season, etc., depending on what makes sense
+        # * Within each gallery, there is one plot for each value in
+        #   'plotParameters', with a corresponding caption and short thumbnail
+        #   description
+        caption = captions[plot_type]
+        write_image_xml(
+            config=self.config,
+            filePrefix=filePrefix,
+            componentName='Ocean',
+            componentSubdirectory='ocean',
+            galleryGroup='Time Series',
+            groupLink='timeseries',
+            gallery='Conservation',
+            thumbnailDescription=title,
+            imageDescription=caption,
+            imageCaption=caption)
+
+    def _derive_variable(self, ds, varname):
+
+        if varname == 'netMassChange':
+            # Convert from kg/s to Gt/s
+            mass_flux = ds['netMassFlux'] * 1e-12
+            # Assume that the frequency of output is monthly
+            dt = constants.sec_per_month
```

Comment:
`sec_per_month` isn't very accurate since it assumes a 30-day month.

Reply:
Is `daysSinceStartOfSim` available in the analysis member?

Reply:
No, it is not available by default. I assumed that `sec_per_month` would at least be the average value in a year so the only error would be from leap years, but I agree that a 30-day month would be too great of an underestimate over 100s-year simulations. I see two options:

1. This change in constants.py: `sec_per_month = sec_per_day * np.mean(daysInMonth)`

2. Changing `buildnml` to include `daysSinceStartOfSim` in a separate PR and then adding an if-clause to use that variable if available

@xylar Should I go ahead and do both or would you prefer one?

Reply:
Sure, I like the first option.  Presumably the answer is just `sec_per_day * 365. / 12.` but you can leave it as you have it.

Reply:
It might be worth checking if `sec_per_month` is used elsewhere in the code, just in case. I doubt it.

Reply:
FYI, It is used in the El Nino spectral analysis. I am making the change anyway because I think that analysis should also benefit from a more accurate month length.